server/controllers/BookController.js

```diff
@@ -75,6 +75,65 @@ class BookController {
       return responseMessage(response, 500, { message: error.message });
     }
   }
+
+  /**
+* Edit Books
+* @name editBook
+* @param {Object} request
+* @param {Object} response
+* @returns {JSON} object
+* @memberof BookController
+*/
+  static async editBook(request, response) {
+    const { id } = request.params;
+    const {
+      title, description, isbn, price, yearPublished, authorId
+    } = request.body;
+    try {
+      const books = await Book.findAll();
+      const availableBook = books.find(book => book.id === +id);
+      if (!availableBook) return responseMessage(response, 400, { message: 'book does not exist' });
+      if (isbn) {
+        const availableIsbn = books.map(book => book.isbn).find(data => data === isbn);
+        if (availableIsbn === isbn) return responseMessage(response, 409, { message: 'isbn already exist' });
+      }
+      const results = await Book.update({
+        title, description, isbn, price, yearPublished
+      }, { where: { id: +id }, returning: true, plain: true });
+      // eslint-disable-next-line no-unused-vars
+      const [numberOfAffectedRows, affectedRows] = await BookAuthor.update({ authorId }, {
+        where: { bookId: +id }, returning: true, plain: true
+      });
+      const author = affectedRows;
+      return responseMessage(response, 200, {
+        status: 'success', message: 'book successfully updated', results: { ...results, author }
```

Comment:
In the response message on your screenshot, the results object in this line isn't showing

Reply:
please check well

Reply:
As at when I made this comment 2 days ago, it wasn't there. Bella also noticed the same thing. You would simply have said that you have fixed it.